riscv/src/register/mvien.rs

```diff
@@ -0,0 +1,160 @@
+//! mvien register
+
+use crate::bits::{bf_extract, bf_insert};
+use riscv_pac::result::{Error, Result};
+use riscv_pac::InterruptNumber;
+
+#[cfg(target_arch = "riscv32")]
+const MASK: usize = 0xffff_e222;
+#[cfg(not(target_arch = "riscv32"))]
+const MASK: usize = 0xffff_ffff_ffff_e222;
+
+read_write_csr! {
+    /// `mvien` register
+    Mvien: 0x308,
+    mask: MASK,
+}
+
+read_write_csr_field! {
+    Mvien,
+    /// Alias of `mie.SSIE`
+    ssoft: 1,
+}
+
+read_write_csr_field! {
+    Mvien,
+    /// Alias of `mie.STIE`
+    stimer: 5,
+}
+
+read_write_csr_field! {
+    Mvien,
+    /// Alias of `mie.SEIE`
+    sext: 9,
+}
+
+impl Mvien {
+    /// Represents the minimum interrupt of the unlabelled virtual interrupt range.
+    pub const MIN_INTERRUPT: usize = 13;
+    /// Represents the maximum interrupt of the unlabelled virtual interrupt range.
+    #[cfg(target_arch = "riscv32")]
+    pub const MAX_INTERRUPT: usize = 31;
+    /// Represents the maximum interrupt of the unlabelled virtual interrupt range.
+    #[cfg(not(target_arch = "riscv32"))]
+    pub const MAX_INTERRUPT: usize = 63;
+
+    /// Gets whether the interrupt number is a valid virtual interrupt.
+    #[inline]
+    pub const fn is_valid_interrupt(int: usize) -> bool {
+        matches!(int, 1 | 5 | 9 | Self::MIN_INTERRUPT..=Self::MAX_INTERRUPT)
+    }
+
+    /// Check if a specific core interrupt source is enabled.
+    ///
+    /// Returns `Error` if the interrupt number is invalid.
+    #[inline]
+    pub fn is_enabled<I: InterruptNumber>(&self, interrupt: I) -> Result<bool> {
```

Comment:
I personally prefer this method like:

```suggestion
    pub fn is_enabled<I: InterruptNumber>(&self, interrupt: I) -> bool {
```

If the interrupt is invalid, it returns `false`, so I guess that is more than enough :)